src/crypto/public_key/elliptic/curves.js

```diff
@@ -169,8 +179,16 @@ class Curve {
     }
   }
 
-  async genKeyPair() {
+  async genKeyPair(plugin) {
     let keyPair;
+    if (this.name === 'webcrypt_p256'){
+      try {
+        return plugin.generateKeyPair(this);
+      } catch (err) {
+        util.printDebugError('Nitrokey WebCrypt failed generating ec key ' + err.message);
+        throw err;
+      }
+    }
```

Comment:
Here we should just check whether the plugin is there.
In addition, this error message is way too specific, that can go inside the plugin function.
And finally, I would prefer not to pass the entire curve object here, if possible, and maybe only the algorithm name and curve name, if that's sufficient?
You can pass the algorithm name from higher up, and get the curve name from `this.name`.

Reply:
Done as suggested